src/cli/repl.rs

```diff
@@ -235,6 +260,30 @@ fn eval_print(
     Ok(ControlFlow::Continue)
 }
 
+
+fn load_module(m: Module, tc_env: &mut TcEnv, codemap: &CodeMap, writer: &StandardStream) {
```

Comment:
What about instead trying to make the API in `lib.rs` more flexible? I'm thinking something like:

```rust
pub fn load_file(tc_env: &mut TcEnv, tc_Env file: &FileMap) -> Result<core::Module, Vec<Diagnostic>> {
```

Would that work?

Reply:
Yup. Well, you could make a copy of the `TcEnv` I guess?

Reply:
Still haven't thought through how this compiler pipeline would look like, thanks for forcing me to think about it! 😅